### PR TITLE
Relative file paths in pkg_path

### PR DESCRIPTION
**Pull request by mfrasco**
# Background information

This PR is responding to [issue 77](https://github.com/UptakeOpenSource/pkgnet/issues/77).

# What it does

This PR allows users to pass `pkg_path` as a relative path instead of just an absolute path. I also clarified documentation and added a unit test to check this new functionality.

# Why it does it

Since, `covr::package_coverage` requires that the directory path passed as an argument is an absolute file path, if a user passes a relative path to `CreatePackageReport`, an error will be thrown that says that the path is invalid. This user experience is frustrating.

# How I did it

I call `tools::file_path_as_absolute` on `pkg_path` within `set_package` of `AbstractPackageReporter`

# Other information

## Manipulation of directory in unit tests

In order to unit test this feature, I changed how the `pkg_path` is passed in the unit tests for `DependencyReporter`. This required manipulation of the working directory. However, I was sure to leave the unit test with the same entry directory.

## Added tools as an imported package

Since this package already uses `tools::package_dependencies`, this PR is not actually introducing a new dependency. However, `tools` was not listed in the `DESCRIPTION` file as an imported package. I made this dependency explicit.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -26,6 +26,7 @@ Imports:
     mvbutils,
     R6,
     rmarkdown,
+    tools,
```
Comment (jameslamb) [inline DESCRIPTION]: oh wow. I wonder how this got past `R CMD CHECK`?

File: R/AbstractPackageReporter.R [hunk 2]
```diff
@@ -37,7 +40,7 @@ AbstractPackageReporter <- R6::R6Class(
             
             if (exists("pkg_path") && !is.null(pkg_path)) {
                 if (dir.exists(pkg_path)) {
-                    private$pkg_path <- pkg_path
+                    private$pkg_path <- tools::file_path_as_absolute(pkg_path)
```
Comment (jameslamb) [inline R/AbstractPackageReporter.R]: 🙌

File: R/PackageFunctionReporter.R [hunk 3]
```diff
@@ -13,9 +13,10 @@
 #'                 previously-set package name, any cached data will be removed.}
 #'             \item{\bold{Args:}}{
 #'                 \itemize{
-#'                 \item{\bold{\code{pkg_name}}: String with the name of the package}
-#'                 \item{\bold{\code{pkg_path}}: Optional path to the source code. 
-#'                     To be used for test coverage, if provided.}
+#'                 \item{\bold{\code{pkg_name}}: String with the name of the package.}
+#'                 \item{\bold{\code{pkg_path}}: Optional directory path to source
```
Comment (jameslamb) [inline R/PackageFunctionReporter.R]: after reading this for the third time, I'm thinking we would benefit from using `inheritParams` to centralize docs for these common parameters. I'll throw up an issue

Comment (jameslamb) [reply R/PackageFunctionReporter.R]: added #81

File: tests/testthat/test-FunctionReporter.R [hunk 4]
```diff
@@ -48,6 +48,7 @@ test_that('FunctionReporter Methods Work', {
       testObj$set_package(
         pkg_name = "baseballstats"
         # Covr only works on source code. find.package path will not work
+        # covr also requires an absolute path, which is provided by system.file
```
Comment (jameslamb) [inline tests/testthat/test-FunctionReporter.R]: good comment, thank you

File: DESCRIPTION (hunk 1 above)
Comment (mfrasco) [reply DESCRIPTION]: Is it because `tools` is a pre-installed package in R?

Comment (jameslamb) [reply DESCRIPTION]: maybe? I've seen it complain about that before though for `methods` (I thought)

who knows. Either way, good change

File: R/AbstractPackageReporter.R (hunk 2 above)
Comment (bburns632) [reply R/AbstractPackageReporter.R]: good catch.  Thanks!

**Pull request merged**